components/Navbar/Navbar.tsx

```diff
@@ -7,6 +7,55 @@ import LogoImage from "../../public/assets/icons/logo-orange.svg";
 import Adapt from "@components/UiKit/Adapt";
 import LinkAsButton from "@components/UiKit/LinkAsButton";
 import * as Styled from "./Navbar.styled";
+import styled from "styled-components";
+import { useRouter } from "next/router";
+
+const SearchWrapper = styled.form<{ searchFocused: boolean }>`
+  width: 100px;
+  transition: width 0.3s ease-in-out;
+  display: flex;
+  align-items: center;
+
+  ${({ searchFocused: searchOpened }) => searchOpened && "width: 300px;"}
```

Comment:
@teaskvarc this is just an example, if you want to do something fancy when the input gets focused